File: advancese7/src/main/java/com/turing/advancese7/tdd/Stack.java
package com.turing.advancese7.tdd;

public class Stack {

	public static final String STACKUNDERFLOW_EXCEPTION = "Stackunderflow exception";
	public static final String STACKOVERFLOW_EXCEPTION = "Stackoverflow exception";
	int SIZE = 10;
	int top = -1;
	int items[] = new int[SIZE];
	
	public void push(int element) {
		if(top < SIZE-1)
		{
			//System.out.println("Top "+top);
			this.items[++top] = element;
		}
		else
		{
			throw new RuntimeException(STACKOVERFLOW_EXCEPTION);
		}
		
	}

	public int size() {
		return top+1;
	}

	public int pop() {
		if(this.top >= 0)
		{
			return items[this.top--];
		}
		else
		{
			throw new RuntimeException(STACKUNDERFLOW_EXCEPTION);
		}
		
	}

}

File: Stack.js
let stack1 ={
    items:[],
    top:0,
}

function push(stack, item){
    stack.items[stack.top] = item;
    stack.top++;
}
function pop(stack){
    if(stack.top === 0)
    {
        return null;
    }
    stack.top--;
    return stack.items[stack.top];
}
push(stack1, 1);
push(stack1, 2);
push(stack1, 3);
console.log(pop(stack1));
//console.log(pop(stack1));
//console.log(pop(stack1));
//console.log(pop(stack1));

stack1.items=[]
push(stack1,100);
console.log(pop(stack1));